bootstrap/cli/remote/session.py

```
@@ -404,9 +511,42 @@ def _check_staged_resource_blobs(root: Path, hash_value: str, issues: list) -> N
+    This is a hard gate — even ``--force`` must not bypass it.
+    """
+    from bootstrap.remote.snapshot import SnapshotStore
+    from bootstrap.remote.verify import Issue
+
+    snap_store = SnapshotStore(root)
+    seen: dict[str, str] = {}
+    for h in session.staged.resources:
+        try:
+            meta, _ = snap_store.load_resource_snapshot(h)
+        except Exception:
+            continue
```

Ruff (0.15.18)

[error] `try`-`except`-`continue` detected, consider logging the exception

(S112)